SignalShareExtension/ShareViewController.swift

```diff
@@ -26,7 +26,66 @@ class ShareViewController: UINavigationController, SAELoadViewDelegate {
 
         _ = AppVersion()
 
-        //[self startupLogging];
+        startupLogging()
+
+        SetRandFunctionSeed()
+
+        // XXX - careful when moving this. It must happen before we initialize TSStorageManager.
+        TSStorageManager.verifyDBKeysAvailableBeforeBackgroundLaunch()
+
+//        // Prevent the device from sleeping during database view async registration
+//        // (e.g. long database upgrades).
+//        //
+//        // This block will be cleared in databaseViewRegistrationComplete.
+//        [DeviceSleepManager.sharedInstance addBlockWithBlockObject:self];
+
+        setupEnvironment()
+
+//        [UIUtil applySignalAppearence];
+//
+//        if (getenv("runningTests_dontStartApp")) {
+//            return YES;
+//        }
```

Comment:
This commented-out code is my next stage.

Reply:
TIOLI: Rather than copying more code from the AppDelegate into this view controller, since this stuff tends to change in key ways, would it make more sense to extract it somewhere accessible to both?

Reply:
Let's revisit that question and see how different they end up.,